vumi/transports/turn/turn.py

```diff
@@ -0,0 +1,205 @@
+# -*- test-case-name: vumi.transports.turn.tests.test_turn -*-
+# -*- encoding: utf-8 -*-
+
+from urllib import urlencode
+from datetime import datetime
+import string
+import warnings
+from StringIO import StringIO
+
+from twisted.web import http
+from twisted.web.resource import Resource
+from twisted.web.server import NOT_DONE_YET
+from twisted.python import log
+from twisted.internet.defer import inlineCallbacks
+from twisted.internet.protocol import Protocol
+from twisted.internet.error import ConnectionRefusedError
+
+from vumi.utils import http_request_full, normalize_msisdn, LogFilterSite
+from vumi.transports.base import Transport
+from vumi.transports.failures import TemporaryFailure, PermanentFailure
+from vumi.errors import VumiError
+
+
+def iso8601(turn_timestamp):
+    # TODO: this will be a unix timestamp
+    if turn_timestamp:
+        ts = datetime.strptime(turn_timestamp, '%Y.%m.%d %H:%M:%S')
+        return ts.isoformat()
+    else:
+        return ''
+
+
+def normalize_outbound_msisdn(msisdn):
+    if msisdn.startswith('+'):
+        return msisdn.replace('+', '')
+    else:
+        return msisdn
+
+class ReceiveEventResource(Resource):
+    isLeaf = True
+
+    def __init__(self, config, publish_event_func, publish_message):
+        self.config = config
+        self.publish_event_func = publish_event_func
+        self.publish_inbound_func = publish_message
+        self.transport_name = 'turn'
+
+    @inlineCallbacks
+    def do_render(self, request):
+        request.setResponseCode(http.CREATED)
+        # request.setHeader('Content-Type', 'text/plain')
+        try:
+            # TODO: validate HMAC secret
+
+            for message in request.args.get("messages", []):
+                message_id = '%s.%s' % (self.transport_name, message["id"])
+                content = ''
+                if message['type'] == 'text':
+                    content = message["text"]["body"]
+                elif message['type'] == 'location':
+                    loc = message["location"]
+                    content = 'geo:{},{}'.format(loc['latitude'], loc['longitude'])
```

Comment:
I think in this case, we rather want to set the content to `None`, and put the location data in the metadata.